### ISSUE TITLE
Smile format out of index bug

### ISSUE THREAD

**Issue by Maxxan** [FasterXML/jackson-dataformat-smile#26]
With latest version (2.6.1) users are starting to get this error:

java.lang.ArrayIndexOutOfBoundsException: length=8000; index=8000
       at com.fasterxml.jackson.dataformat.smile.SmileParser._findDecodedFromSymbols(SmileParser.java:1898)
       at com.fasterxml.jackson.dataformat.smile.SmileParser._handleFieldName(SmileParser.java:1521)
       at com.fasterxml.jackson.dataformat.smile.SmileParser.nextToken(SmileParser.java:587)

**Comment by cowtowncoder** [comment]
Thank you for reporting this. A reproducible test case would be great, if possible.
I am unfortunately (wrt Smile module, not me :) ) on vacation, and will be returning next week.
But once I get back I should be able to fix this, and get 2.6.2 update (or, depending, 2.6.1-1 micro-patch) out very soon.

**Comment by cowtowncoder** [comment]
Ok, I can see where the problem occurs. I could have sworn I just fixed this problem, or something similar... need to see what gives.

Ah. Actually, it's code shared with CBOR codec, and fix for that was:

https://github.com/FasterXML/jackson-dataformat-cbor/pull/13

but I hadn't seen that in Smile codec, which I thought I had checked. Apparently I did not.
Will try to reproduce with Smile, fix for 2.6.2.
